Source/missiles.cpp

```diff
@@ -4566,14 +4668,15 @@ void missiles_process_charge()
 		} else {
 			graphic = MonsterGraphic::Walk;
 		}
-		Missiles[mi]._miAnimData = mon->GetAnimData(graphic).CelSpritesForDirections[mis->_mimfnum]->Data();
+		missile._miAnimData = mon->GetAnimData(graphic).CelSpritesForDirections[mis->_mimfnum]->Data();
 	}
 }
 
 void ClearMissileSpot(int mi)
 {
-	dFlags[Missiles[mi].position.tile.x][Missiles[mi].position.tile.y] &= ~BFLAG_MISSILE;
-	dMissile[Missiles[mi].position.tile.x][Missiles[mi].position.tile.y] = 0;
+	auto &missile = Missiles[mi];
+	dFlags[missile.position.tile.x][missile.position.tile.y] &= ~BFLAG_MISSILE;
+	dMissile[missile.position.tile.x][missile.position.tile.y] = 0;

```

Comment:
```suggestion
	auto &missileTile = Missiles[mi].position.tile;
	dFlags[missileTile.x][missileTile.y] &= ~BFLAG_MISSILE;
	dMissile[missileTile.x][missileTile.y] = 0;
```
This only needs the dungeon tile, can go deeper into the struct immediately.

The function could take a Point as the argument instead of an index as well, it's called from a context where we've already accessed the relevant item in the Missiles array.